hta/common/trace_parser.py

```diff
@@ -428,3 +436,102 @@ def parse_trace_dataframe(
             f"Parser Memory usage peak = {(peak/1024/1024):.2f} MB, current = {(current/1024/1024):.2f} MB"
         )
     return meta, df, local_symbol_table
+
+
+# --- Trace metadata reader ---
+def parse_metadata_ijson(fh) -> MetaData:
+    """
+    Parses a trace file using the `ijson` library and extracts certain high-level key-value pairs.
+    Args:
+        fh (file-like object): The trace file to parse.
+    Returns:
+        A dictionary containing the extracted metadata.
+
+    Note: We use the ijson low level API to read all dictionary
+    elements in the json upto "traceEvents", at which point we just terminate.
+    This leads to some blazingly fast metadata parsing!
+
+    https://pypi.org/project/ijson/#toc-entry-4
+
+    The ijson low-level API has a series of (prefix, event, value). The
+    prefix shows the path in the json object, while event can signify
+    the start of a map, array or a key. Following is an excerpt of events for a trace file
+
+    ```
+     start_map None
+     map_key schemaVersion
```

Comment:
a little confused with these because there are only two values. from the comment I thought there should be three values (prefix, event, value) for each?

Reply:
oh yea here prefix = '', this is the case for everything at root level.